src/components/fields/location-input-group/location-modal/location-search/location-search.tsx

```diff
@@ -0,0 +1,749 @@
+	const [currentPaginationPageNum, setCurrentPaginationPageNum] = useState(1);
+	// =============================================================================
+	// EFFECTS
+	// =============================================================================
+	useEffect(() => {
+		// if only address
+		if ((formValues?.address && !formValues?.lat) || !formValues?.lng) {
+			(async () => {
+				try {
+					const res = await OneMapService.searchByAddress({
+						searchValue: formValues.address,
+						returnGeom: OneMapBoolean.YES,
+						getAddressDetails: OneMapBoolean.YES,
+						pageNum: 1,
+					});
+					const postalCodeCheck =
+						!mustHavePostalCode || LocationHelper.hasGotAddressValue(res.results?.[0].POSTAL);
+					if (res.results?.[0] && postalCodeCheck) {
+						// Get full address
+						const fetchedFormValue = {
+							address: formValues.address || "",
+							blockNo: res.results?.[0].BLK_NO,
+							building: res.results?.[0].BUILDING,
+							postalCode: res.results?.[0].POSTAL,
+							roadName: res.results?.[0].ROAD_NAME,
+							lat: parseFloat(res.results?.[0].LATITUDE) || undefined,
+							lng: parseFloat(res.results?.[0].LONGITUDE) || undefined,
+							x: parseFloat(res.results?.[0].X) || undefined,
+							y: parseFloat(res.results?.[0].Y) || undefined,
+						};
+						updateFormValues(fetchedFormValue);
+						changeSelectedAddressInfo(fetchedFormValue);
+						// populate result list
+						setQueryString(fetchedFormValue.address);
+						setUseQuerySearch(true);
+					} else {
+						updateFormValues({});
+						changeSelectedAddressInfo({});
+					}
+				} catch (error) {}
+			})();
+			// if only latlng
+		} else if (reverseGeoCodeEndpoint && !formValues?.address && formValues?.lat && formValues?.lng) {
+			(async () => {
+				try {
+					const locationList = await OneMapService.reverseGeocode({
+						route: reverseGeoCodeEndpoint,
+						latitude: formValues.lat,
+						longitude: formValues.lng,
+					});
+					const postalCodeCheck =
+						!mustHavePostalCode || LocationHelper.hasGotAddressValue(locationList?.[0].POSTALCODE);
+					if (locationList?.[0] && postalCodeCheck) {
+						const address = LocationHelper.formatAddressFromGeocodeInfo(locationList?.[0], true);
+						// Get full address
+						const fetchedFormValue = {
+							address,
+							blockNo: locationList[0].BLOCK,
+							building: locationList[0].BUILDINGNAME,
+							postalCode: locationList[0].POSTALCODE,
+							roadName: locationList[0].ROAD,
+							lat: formValues.lat,
+							lng: formValues.lng,
+							x: parseFloat(locationList[0].XCOORD) || undefined,
+							y: parseFloat(locationList[0].YCOORD) || undefined,
+						};
+						updateFormValues(fetchedFormValue);
+						changeSelectedAddressInfo(fetchedFormValue);
+						// populate result list
+						setQueryString(fetchedFormValue.address);
+						setUseQuerySearch(true);
+					}
+				} catch (error) {}
+			})();
+		}
+	}, []);
```

Comment:
understand that this is moved here because of the intention to centralise all the api calls in this file, however the need to update form values here looks clunky, may I request to:
- move the api calls to the new helpers.ts
- combine the `searchByAddress` call with `fetchAddress()` helper function
- these new helper functions should also format the values into the objects before returning
- move this parsing of default values and setting of form values back to the main `location-field` component (sorry!)